e2e-tests/tests/test_substrate_smoke.py

```diff
@@ -0,0 +1,151 @@
+
+        # Use substrate interface's built-in transfer method to avoid custom transaction building issues
+        try:
+            # Create call manually
+            call = api.substrate.compose_call(
+                call_module='Balances',
+                call_function='transfer_allow_death',
+                call_params={
+                    'dest': new_wallet.address,
+                    'value': value
+                }
+            )
+            
+            # Create and submit signed extrinsic directly
+            signed_extrinsic = api.substrate.create_signed_extrinsic(
+                call=call, 
+                keypair=sender_wallet.raw
+            )
+            
+            receipt = api.substrate.submit_extrinsic(
+                signed_extrinsic, 
+                wait_for_finalization=True
+            )
+            
+            logger.info(f"Transaction hash: {receipt.extrinsic_hash}")
+            logger.info(f"Transaction fee: {receipt.total_fee_amount}")
+            
+            # Check final balances
+            sender_balance_after = api.get_pc_balance(sender_wallet.address)
+            receiver_balance_after = api.get_pc_balance(new_wallet.address)
+            
+            logger.info(f"Sender balance after: {sender_balance_after}")  
+            logger.info(f"Receiver balance after: {receiver_balance_after}")
+
+            # Verify balances (relaxed verification since we can't predict exact fees)
+            assert receiver_balance_after >= receiver_balance_before + value, \
+                f"Receiver should have at least {receiver_balance_before + value}, got: {receiver_balance_after}"
+            
+            assert sender_balance_after < sender_balance_before, \
+                f"Sender balance should have decreased from {sender_balance_before}, got: {sender_balance_after}"
+                
+        except Exception as e:
+            error_msg = str(e)
+            logger.error(f"Transaction failed: {e}")
+            
+            # Log transaction error details for debugging
+            logger.error(f"Transaction failed: {error_msg}")
+            raise
+
+    @mark.test_key('SUBSTRATE-003')
+    def test_node_info(self, api: BlockchainApi):
+        """Test basic node info retrieval
+        
+        * get system chain name
+        * get system version
+        * get system health
+        * verify responses are valid
+        """
+        # Get system info using substrate interface
+        chain_name = api.substrate.rpc_request("system_chain", [])
+        version = api.substrate.rpc_request("system_version", [])  
+        health = api.substrate.rpc_request("system_health", [])
+        
+        logger.info(f"Chain: {chain_name}")
+        logger.info(f"Version: {version}")
+        logger.info(f"Health: {health}")
+        
+        # Basic assertions
+        assert chain_name is not None, "Chain name should not be None"
```

Comment:
Better place assertions before logging. Thus if any value is None - logs will not be populated with it.